src/task/SubscribeAndReceiveSubscribeAckTask.ts

```diff
@@ -176,22 +192,73 @@ export default class SubscribeAndReceiveSubscribeAckTask extends BaseTask {
       if (section.direction === 'recvonly') {
         const streamId = midsToStreamIds.get(section.mid);
         if (streamId === undefined) {
-          this.context.logger.warn(`Could not find stream ID for MID: ${section.mid}`);
+          this.logger.warn(`Could not find stream ID for MID: ${section.mid}`);
           continue;
         }
         newSubscriptions.push(streamId);
       } else {
         newSubscriptions.push(0);
       }
     }
-    this.context.logger.info(
+    this.logger.info(
       `Fixed up ${JSON.stringify(videoSubscriptions)} to ${JSON.stringify(
         newSubscriptions
       )} (may be same))}`
     );
     return newSubscriptions;
   }
 
+  private convertVideoPreferencesToVideoSubscriptionConfiguration(
+    receiveStreamIds: number[],
+    preferences: VideoPreferences
+  ): SignalingClientVideoSubscriptionConfiguration[] {
+    if (
+      this.context.transceiverController.getMidForStreamId === undefined ||
+      preferences === undefined
+    ) {
+      return [];
+    }
+
+    const configurations = new Array<SignalingClientVideoSubscriptionConfiguration>();
+    const attendeeIdToMid = new Map<string, string>();
+    const attendeeIdToGroupId = new Map<string, number>();
+    for (const streamId of receiveStreamIds) {
+      // The local description will have been set by the time this task is running, so all
+      // of the transceivers should have `mid` set by now (see comment above `getMidForStreamId`)
+      const mid = this.context.transceiverController.getMidForStreamId(streamId);
+      if (mid === undefined) {
+        // Likely imposible to hit the next `if` given use of `fixUpSubscriptionOrder`
+        /* istanbul ignore if */
+        if (streamId !== 0) {
+          // Send section or inactive section
+          this.context.logger.warn(`Could not find MID for stream ID: ${streamId}`);
+        }
+        continue;
+      }
+      const attendeeId = this.context.videoStreamIndex.attendeeIdForStreamId(streamId);
+      attendeeIdToMid.set(attendeeId, mid);
+      attendeeIdToGroupId.set(
+        attendeeId,
+        this.context.videoStreamIndex.groupIdForStreamId(streamId)
+      );
+    }
+    for (const preference of preferences) {
+      const configuration = new SignalingClientVideoSubscriptionConfiguration();
+      const mid = attendeeIdToMid.get(preference.attendeeId);
+      if (mid === undefined) {
+        this.context.logger.warn(`Could not find MID for attendee ID: ${preference.attendeeId}`);
+        continue;
+      }
+      configuration.mid = mid;
+      configuration.attendeeId = preference.attendeeId;
+      configuration.groupId = attendeeIdToGroupId.get(preference.attendeeId);
+      configuration.priority = Number.MAX_SAFE_INTEGER - preference.priority;
```

Comment:
Maybe we need a comment in the proto or here to explain the logic of priority in the backend?

Reply:
Sure. My intention was to avoid the classic "This video was set to high priority...by setting it to a low value" statement which always confuses people lol.

Reply:
But if it is new in the backend, wouldn't it better to be consistent?

Reply:
Discussed offline.